### DISCUSSION TITLE
Simple and crude tab-completion for pattern names

### DISCUSSION THREAD

**Discussion by residualmind** [danielmiessler/Fabric#312] [Ideas]
Before I discovered that all these nice aliases are being created, while I was playing with the existing patterns, I really wanted a shell tab completion for the pattern names. This is a very simple way of achieving it:

Running

`complete -W "$(ls ~/.config/fabric/patterns)" fabric`

 gives you a very simple but effective TAB-completion for pattern names in bash and zsh.

Is this against the philosophy of using aliases? If not, it could be added to the bootstrap file, I'd be happy to create a PR.

**Comment by ksylvan** [comment]
Shell completions for Fabric has been implemented for a while now.

You can even install it using a one-line curl command::

https://github.com/danielmiessler/Fabric?tab=readme-ov-file#quick-install-no-clone-required
